src/core/thread/router_table.cpp

```diff
@@ -881,6 +881,8 @@ void RouterTable::HandleTableChanged(void)
 #if OPENTHREAD_CONFIG_HISTORY_TRACKER_ENABLE
     Get<Utils::HistoryTracker>().RecordRouterTableChange();
 #endif
+
+    Get<Mle::MleRouter>().RecalculateAdvertiseInterval();
```

Comment:
Suppose that the Adv Trickle timer was not running (for whatever reason - it seems to be possible, looking at the code) and an event comes in that the router table is changed.  Is it really intended here to start the Trickle timer now? (which is a side effect of `RecalculateAdvertiseInterval()` )
If yes, why wasn't it started here in the original code?

Reply:
I took care of this in new PR -> https://github.com/openthread/openthread/pull/9307.